[question] Struggling with update to 0.14

Hi,
I'm currently trying to migrate my existing project to 0.14 so that I can use the new macros, but I'm struggling with my ZipWriter implementation. KotoObject now implies KotoSync and that seems to break some things. Here's my Zip struct:
```rust
#[derive(Clone, KotoCopy, KotoType)]
struct Zip {
    inner: PtrMut<ZipWriter<File>>,
    needs_finish: bool,
}
```

This gives me the following error when I try to create a koto_impl for it:
```
error[E0277]: `NonNull<zstd_sys::ZSTD_CCtx_s>` cannot be shared between threads safely
   --> src\scripting\api\arch.rs:181:21
    |
181 | impl KotoObject for Zip {
    |                     ^^^ `NonNull<zstd_sys::ZSTD_CCtx_s>` cannot be shared between threads safely
    |
    = help: within `ZipWriter<std::fs::File>`, the trait `KotoSync` is not implemented for `NonNull<zstd_sys::ZSTD_CCtx_s>`, which is required by `api::arch::Zip: KotoSync`
note: required because it appears within the type `zstd_safe::CCtx<'static>`
   --> C:\Users\asaljo\.cargo\registry\src\index.crates.io-6f17d22bba15001f\zstd-safe-5.0.2+zstd.1.5.2\src\lib.rs:156:12
    |
156 | pub struct CCtx<'a>(NonNull<zstd_sys::ZSTD_CCtx>, PhantomData<&'a ()>);
    |            ^^^^
note: required because it appears within the type `zstd::stream::raw::Encoder<'static>`
```

My program is single-threaded anyway so I thought maybe switching the underlying PtrMut implementation from Arc to Rc would help but I didn't manage to use the rc feature instead of arc (always getting the double-feature warning from koto_memory, although I disabled default-features).

Any help would be appreciated.

Even with these dependencies (note explicit `koto_memory` reference)
```
koto = { version = "0.14.0", default-features = false, features = ["rc"] }
koto_json = { version = "0.14.0", default-features = false, features = ["rc"] }
koto_memory = { version = "0.14.0", default-features = false, features = ["rc"] }
koto_regex = { version = "0.14.0", default-features = false, features = ["rc"] }
koto_serialize = { version = "0.14.0", default-features = false, features = ["rc"] }
koto_tempfile = { version = "0.14.0", default-features = false, features = ["rc"] }
koto_toml = { version = "0.14.0", default-features = false, features = ["rc"] }
koto_yaml = { version = "0.14.0", default-features = false, features = ["rc"] }
```
I get redefinition errors from koto_memory:
```
error[E0428]: the name `make_ptr` is defined multiple times
  --> C:\Users\asaljo\.cargo\registry\src\index.crates.io-6f17d22bba15001f\koto_memory-0.14.0\src\rc\ptr.rs:18:1
   |
18 | macro_rules! make_ptr {
   | ^^^^^^^^^^^^^^^^^^^^^ `make_ptr` redefined here
   |
  ::: C:\Users\asaljo\.cargo\registry\src\index.crates.io-6f17d22bba15001f\koto_memory-0.14.0\src\arc\ptr.rs:18:1
   |
18 | macro_rules! make_ptr {
   | --------------------- previous definition of the macro `make_ptr` here
   |
   = note: `make_ptr` must be defined only once in the macro namespace of this module
```
and this error:
```
error: A single memory management feature can be enabled at a time
  --> C:\Users\asaljo\.cargo\registry\src\index.crates.io-6f17d22bba15001f\koto_memory-0.14.0\src\lib.rs:16:1
   |
16 | compile_error!("A single memory management feature can be enabled at a time");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -43,6 +43,34 @@ jobs:
       - name: Test with optimizations
         run: just test_release --verbose
 
+  build_and_test_rc:
+    # We don't need to test rc builds on all platforms for now
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: just@1.5.0
+
+      - name: Test the rc build variant
+        run: just test_rc --verbose
+
+  build_and_test_release_rc:
+    # We don't need to test rc builds on all platforms for now
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: just@1.5.0
+
+      - name: Test the rc build variant
+        run: just test_rc --release --verbose
+
   code_checks:
     runs-on: ubuntu-latest
 

--- a/justfile
+++ b/justfile
@@ -57,7 +57,6 @@ test_parser:
 
 test_release *args:
   just test --release {{args}}
-  just test_rc --release {{args}}
 
 test_runtime:
   cargo test --package koto_runtime

--- a/justfile
+++ b/justfile
@@ -26,7 +26,10 @@ test *args:
   cargo test {{args}}
 
 test_rc *args:
-  cargo test -p koto_runtime --no-default-features --features rc {{args}}
+  cargo test  --no-default-features --features rc \
+    -p koto_runtime \
+    -p lib_tests \
+    {{args}}
 
 test_benches:
   cargo test --benches

--- a/libs/json/Cargo.toml
+++ b/libs/json/Cargo.toml
@@ -15,11 +15,15 @@ arc = ["koto_runtime/arc"]
 rc = ["koto_runtime/rc"]
 
 [dependencies]
-koto_serialize = { path = "../../crates/serialize", version = "^0.15.0" }
 serde_json = { workspace = true }
 
 [dependencies.koto_runtime]
 path = "../../crates/runtime"
+version = "^0.15.0"
+default-features = false
+
+[dependencies.koto_serialize]
+path = "../../crates/serialize"
 version = "^0.15.0"
 default-features = false
 

--- a/libs/lib_tests/Cargo.toml
+++ b/libs/lib_tests/Cargo.toml
@@ -9,13 +9,18 @@ description = "A test runner for the standard Koto libraries"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+[features]
+default = ["arc"]
+arc = ["koto/arc"]
+rc = ["koto/rc"]
+
 [dev-dependencies]
-koto = { path = "../../crates/koto", version = "^0.15.0" }
-koto_color = { path = "../color", version = "^0.15.0" }
-koto_geometry = { path = "../geometry", version = "^0.15.0" }
-koto_json = { path = "../json", version = "^0.15.0" }
-koto_random = { path = "../random", version = "^0.15.0" }
-koto_regex = { path = "../regex", version = "^0.15.0" }
-koto_tempfile = { path = "../tempfile", version = "^0.15.0" }
-koto_toml = { path = "../toml", version = "^0.15.0" }
-koto_yaml = { path = "../yaml", version = "^0.15.0" }
+koto = { path = "../../crates/koto", version = "^0.15.0", default-features = false }
+koto_color = { path = "../color", version = "^0.15.0", default-features = false }
+koto_geometry = { path = "../geometry", version = "^0.15.0", default-features = false }
+koto_json = { path = "../json", version = "^0.15.0", default-features = false }
+koto_random = { path = "../random", version = "^0.15.0", default-features = false }
+koto_regex = { path = "../regex", version = "^0.15.0", default-features = false }
+koto_tempfile = { path = "../tempfile", version = "^0.15.0", default-features = false }
+koto_toml = { path = "../toml", version = "^0.15.0", default-features = false }
+koto_yaml = { path = "../yaml", version = "^0.15.0", default-features = false }

--- a/libs/toml/Cargo.toml
+++ b/libs/toml/Cargo.toml
@@ -15,11 +15,15 @@ arc = ["koto_runtime/arc"]
 rc = ["koto_runtime/rc"]
 
 [dependencies]
-koto_serialize = { path = "../../crates/serialize", version = "^0.15.0" }
 toml = { workspace = true }
 
 [dependencies.koto_runtime]
 path = "../../crates/runtime"
+version = "^0.15.0"
+default-features = false
+
+[dependencies.koto_serialize]
+path = "../../crates/serialize"
 version = "^0.15.0"
 default-features = false
 

--- a/libs/yaml/Cargo.toml
+++ b/libs/yaml/Cargo.toml
@@ -15,11 +15,15 @@ arc = ["koto_runtime/arc"]
 rc = ["koto_runtime/rc"]
 
 [dependencies]
-koto_serialize = { path = "../../crates/serialize", version = "^0.15.0" }
 serde_yaml = { workspace = true }
 
 [dependencies.koto_runtime]
 path = "../../crates/runtime"
+version = "^0.15.0"
+default-features = false
+
+[dependencies.koto_serialize]
+path = "../../crates/serialize"
 version = "^0.15.0"
 default-features = false
 

